### ISSUE TITLE
Smoothing of a constant plot is messed up

### ISSUE THREAD

**Issue by nikonikolov** [tensorflow/tensorboard#786]
When I try to smooth a value which is constant, then some strange spikes appear in the plot. 
For example: 
![image](https://user-images.githubusercontent.com/11044035/33515880-fc3d65dc-d768-11e7-8c72-0afb43299d8f.png)
Here `actor_learn_rate` is always a constant. Interestingly, in its own tab it is displayed correctly, but in the Tags matching tab it is messed up. When I hover over the plot at a point corresponding to a spike, the correct value of `1e-4` is reported. The case is the same for `critic_learn_rate`.

Interestingly, the spikes depend on the smoothing value. For example, for a smoothing value of `0.222`, there are no spikes:
![image](https://user-images.githubusercontent.com/11044035/33515919-a1bab38e-d769-11e7-8db6-7c6c12ce6c67.png)
However, the orange line is not displayed at `1e-4` which is the correct value. 

Then a smoothing value of `0` is the same.
![image](https://user-images.githubusercontent.com/11044035/33515857-d4d54f3c-d768-11e7-92a3-f55290058882.png)

My version is `tensorflow-tensorboard (0.4.0rc3)`

**Comment by netheril96** [comment]
I have the same problem. It only shows up in the recent version the tensorboard, but not old ones.

**Comment by nfelt** [comment]
Hmm, if it's only showing up in the latest release of TensorBoard, I wonder if it might be inadvertently introduced by https://github.com/tensorflow/tensorboard/pull/639 ?  cc @alexirpan

**Comment by danijar** [comment]
I can try to take a look at the smoothing code (there are some other changes I'd like to play with). Can I edit the smoothing code of a running TensorBoard instance from the browser somehow or how would you suggest getting started?

**Comment by alexirpan** [comment]
When I worked on #639 I downloaded + built a local copy of the Tensorboard codebase, then worked off of that.

I suspect it's some issue with floating point error, where the code does

    d.smoothed = last / debiasWeight;

**Comment by bileschi** [comment]
@stephanwlee thanks for your recent fixes!  Can this issue be closed?

**Comment by nfelt** [comment]
Per @wchargin comments on the fix PR I think we consider the general problem here to still not be solved: https://github.com/tensorflow/tensorboard/pull/1698#issuecomment-448066849

**Comment by wchargin** [comment]
Correct; what’s needed is a more numerically stable implementation of
the smoothing algorithm. This is “contributions welcome” under our usual
policy for smoothing algorithm changes:

> It seems that every change in the smoothing algorithm results in new
> tradeoffs and biases, and new requests to change the algorithm. At
> this point it's not a priority for the team, so just filing an issue
> probably won't directly result in changes. However, the smoothing
> algorithm is actually a very simple piece of code to change. If you
> come up with a new algorithm, and demonstrate persuasively why it is
> an improvement over the current one (taking into account new biases
> and edge cases that it introduces), \[we’ll\] happily merge it.

(<https://github.com/tensorflow/tensorboard/issues/610#issuecomment-334380578>)

**Comment by danijar** [comment]
I'd still be interested in playing with the smoothing algorithm. Same question as a while ago:

> I can try to take a look at the smoothing code (there are some other changes I'd like to play with). Can I edit the smoothing code of a running TensorBoard instance from the browser somehow or how would you suggest getting started?

**Comment by wchargin** [comment]
There’s not an easy way to edit the smoothing algorithm live in the
browser, no. You’ll have to restart the server after each change to the
logic.

I suppose that it’s possible that you could change the smoothing
algorithm to `eval` a global string variable that you modify in the
browser, but I’ve never tried this. :-)

**Comment by pindinagesh** [comment]
@nikonikolov 

Could you please move this to closed status as the [PR](https://github.com/tensorflow/tensorboard/pull/1698#issuecomment-448066849) already merged. Thanks

**Comment by nfelt** [comment]
@pindinagesh the PR doesn't fully address the issue, which I explained in https://github.com/tensorflow/tensorboard/issues/786#issuecomment-568036820

So this issue should stay open pending a more general solution.
